paper-tooltip.html

```diff
@@ -215,30 +236,46 @@
           this.unlisten(this._target, 'focus', 'show');
           this.unlisten(this._target, 'mouseleave', 'hide');
           this.unlisten(this._target, 'blur', 'hide');
+          this.unlisten(this._target, 'tap', 'hide');
+          this.unlisten(this, 'mouseenter', 'hide');
         }
       },
 
       show: function() {
+        // If the tooltip is already showing, there's nothing to do.
         if (this._showing)
           return;
 
         if (Polymer.dom(this).textContent.trim() === '')
           return;
 
-        this.cancelAnimation();
 
+        this.cancelAnimation();
+        this._showing = true;
         this.toggleClass('hidden', false, this.$.tooltip);
         this.updatePosition();
-        this._showing = true;
 
+        this.animationConfig.entry[0].timing.delay = this.animationDelay;
+        this._animationPlaying = true;
         this.playAnimation('entry');
       },
 
       hide: function() {
-        if (!this._showing)
+        // If the entry animation is still playing, don't try to play the exit
+        // animation since this will reset the opacity to 1. Just end the animation.
+        if (this._showing && this._animationPlaying) {
+          this.cancelAnimation();
+          this._showing = false;
+          this._onAnimationFinish();
```

Comment:
It seems like you always intend to return early in this case. Perhaps just `return` here?